Added Swap X and Metropolis

okolicodes7 opened the pull request:
## Thank you for contributing to Spellbook 🪄
Please open the PR in **draft** and mark as ready when you want to request a review. 

### Description:
Added Swap X and Metropolis to the Sonic DEX section
---
quick links for more information:
- [README.md](https://github.com/duneanalytics/spellbook/blob/main/README.md)
- [spellbook docs](https://github.com/duneanalytics/spellbook/tree/main/docs)
- [CONTRIBUTING.md](https://github.com/duneanalytics/spellbook/blob/main/CONTRIBUTING.md)

dune-eng commented:
Workflow run id 15858136454 approved.

dune-eng commented:
Workflow run id 15874558805 approved.

dune-eng commented:
Workflow run id 15874558814 approved.

dune-eng commented:
Workflow run id 15874559281 approved.

okolicodes7 commented:
@jeff-dude I've made the additions as you directed. Pls check it out. Thanks

jeff-dude commented:
> @jeff-dude I've made the additions as you directed. Pls check it out in the second commit i made. Thanks

looks like you are establishing the wrong sources. please see the attached GH workflow failures for CI testing. the models can't find the right sources you supply to the macros.

dune-eng commented:
Workflow run id 15937323486 approved.

dune-eng commented:
Workflow run id 15937323485 approved.

okolicodes7 commented:
@jeff-dude I couldn't find any swap events for metropolis as i looked thru dune and only found swap function calls. AS it's my first time I kinda mirrored existing dex tables when doing the additions so i just deleted the swap function call in the source file.

dune-eng commented:
Workflow run id 15937898261 approved.

dune-eng commented:
Workflow run id 15937898266 approved.

dune-eng commented:
Workflow run id 15937898649 approved.

dune-eng commented:
Workflow run id 15937933645 approved.

dune-eng commented:
Workflow run id 15937933523 approved.

dune-eng commented:
Workflow run id 15937933518 approved.

dune-eng commented:
Workflow run id 15938002580 approved.

dune-eng commented:
Workflow run id 15938002581 approved.

dune-eng commented:
Workflow run id 15938002673 approved.

jeff-dude commented on dbt_subprojects/dex/models/trades/sonic/platforms/metropolis_sonic_base_trades.sql:
```diff
@@ -0,0 +1,21 @@
+        blockchain = 'sonic',
+        project = 'metropolis',
+        version = '1',
+        Factory_evt_function = source('metropolis_sonic', 'router_call_swapexacttokensfortokens'),
```
this is your error. there is no variable named `Factory_evt_function`. you need to use the supplied ones in the uni v2 macro
https://github.com/duneanalytics/spellbook/blob/main/dbt_subprojects/dex/macros/models/_project/uniswap_compatible_trades.sql#L1-L9

jeff-dude replied:
however, the existing macro won't accept function decoded tables. it's a query built explicitly for events. if metropolis doesn't have events, then you need to build the model as a standalone query and not use the macro

jeff-dude replied:
plz don't change the macro to fit this use case, as it's heavily used elsewhere

okolicodes7 replied:
Pls @jeff-dude after building the model, where should i save it? Which folder etc

jeff-dude reviewed (changes requested):
i highly recommend using `dbt compile` locally and grabbing raw sql output in target directory of your path to model(s). then copy/paste and run on dune app, so you can test the model query prior to running on CI and failing with unknowns.

dune-eng commented:
Workflow run id 16037206551 approved.

jeff-dude commented on dbt_subprojects/dex/models/trades/sonic/platforms/metropolis_sonic_base_trades.sql:
```diff
@@ -0,0 +1,23 @@
+
+
+   {{
+    metropolis_compatible_trades(
```
there is no indication that this metropolis logic will be used heavily by other DEXs/models in the future. the macro includes a harcoded CTE which is specific to metropolis on sonic. let's remove the need for a macro and put that SQL logic in this model file directly.

jeff-dude replied:
yes, move code from macro to model, delete macro.
based on your error there, it looks like you did not remove reference to macro in the model so it's failing while looking for it, which was deleted. there should be no macro call.

jeff-dude commented:
> @jeff-dude pls check it now..

please fix merge conflicts and get CI workflow running. right now, no code has run, so it's hard to review completely.

dune-eng commented:
Workflow run id 16237243499 approved.

dune-eng commented:
Workflow run id 16237243494 approved.

dune-eng commented:
Workflow run id 16237243579 approved.

dune-eng commented:
Workflow run id 16245818305 approved.

dune-eng commented:
Workflow run id 16245818296 approved.

dune-eng commented:
Workflow run id 16245818491 approved.

okolicodes7 commented:
@jeff-dude I fixed the merge conflicts and removed the macro as you said and i keep running into issues 
``` 
Compilation Error
  Model 'model.dex.metropolis_sonic_base_trades' (models/trades/sonic/platforms/metropolis_sonic_base_trades.sql) depends on a source named 'metropolis_sonic.router_call_swapexacttokensforeth' which was not found
05:17:52  [WARNING][DeprecationsSummary]: Deprecated functionality
Summary of encountered deprecations:
- DuplicateYAMLKeysDeprecation: 1 occurrence
- PropertyMovedToConfigDeprecation: 12 occurrences
To see all deprecation instances instead of just the first occurrence of each,
run command again with the `--show-all-deprecations` flag. You may also need to
run with `--no-partial-parse` as some deprecations are only encountered during
parsing.
```
If you look at the code i just pushed, the source is contained properly and should work fine. Will really appreciate your feedback.

dune-eng commented:
Workflow run id 16291329121 approved.

dune-eng commented:
Workflow run id 16291329123 approved.

dune-eng commented:
Workflow run id 16291330150 approved.

okolicodes7 commented:
@jeff-dude i've made the corrections

jeff-dude commented:
CI pipeline is still failing. run `dbt compile` locally, grab both model queries, run on dune app and you'll see errors. fix until they run on the dune app.

okolicodes7 commented:
That’s been my issue. Been trying to run the dbt compile and I’m getting the error I showed in my previous comment

okolicodes7 commented:
Good day @jeff-dude As show in the output below, it shows the correct profile config: 
```
11:49:53  Using profiles dir at /home/okolicodes/Dunespellbook/spellbook/dbt_subprojects/dex
11:49:53  Using profiles.yml file at /home/okolicodes/Dunespellbook/spellbook/dbt_subprojects/dex/profiles.yml
11:49:53  Using dbt_project.yml file at /home/okolicodes/Dunespellbook/spellbook/dbt_subprojects/dex/dbt_project.yml
11:49:53  adapter type: trino
11:49:53  adapter version: 1.9.2
11:49:53  Configuration:
11:49:53    profiles.yml file [OK found and valid]
11:49:53    dbt_project.yml file [OK found and valid]
11:49:53  Required dependencies:
11:49:53   - git [OK found]

11:49:53  Connection:
11:49:53    host: trino
11:49:53    port: 1234
11:49:53    user: trino
11:49:53    database: hive
11:49:53    schema: wizard
11:49:53    cert: None
11:49:53    prepared_statements_enabled: True
11:49:53  Registered adapter: trino=1.9.2
```
But it still fails here: 
```
INFO:trino.client:failed after 3 attempts
11:50:24    Connection test: [ERROR]

11:50:24  1 check failed:
11:50:24  dbt was unable to connect to the specified database.
The database returned the following error:

  >Database Error
  failed to execute: HTTPConnectionPool(host='trino', port=1234): Max retries exceeded with url: /v1/statement (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x7f10b21f75f0>: Failed to resolve 'trino' ([Errno -3] Temporary failure in name resolution)"))
```

jeff-dude commented:
i'm not sure what could be happening. i haven't seen users hit this issue before. `dbt compile` doesn't try to connect to a server, so something in your local setup is off. if you can't get that resolved, then just keep going through the CI pipeline attached to the PR here and iterate on changes through the log outputs there. it is currently failing on bad union.

dune-eng commented:
Workflow run id 16462250429 approved.

dune-eng commented:
Workflow run id 16463011611 approved.

dune-eng commented:
Workflow run id 16463387518 approved.

dune-eng commented:
Workflow run id 16464782814 approved.

dune-eng commented:
Workflow run id 16464782854 approved.

dune-eng commented:
Workflow run id 16464783344 approved.

dune-eng commented:
Workflow run id 16485351051 approved.

dune-eng commented:
Workflow run id 16485351061 approved.

dune-eng commented:
Workflow run id 16485596939 approved.

dune-eng commented:
Workflow run id 16485596923 approved.

dune-eng commented:
Workflow run id 16485648994 approved.

dune-eng commented:
Workflow run id 16485648986 approved.

dune-eng commented:
Workflow run id 16485649150 approved.

okolicodes7 commented:
`(models/trades/sonic/platforms/metropolis_sonic_base_trades.sql)
  TrinoUserError(type=USER_ERROR, name=SCHEMA_NOT_FOUND, message="line 101:30: Schema 'git_dunesql_dc552a1' does not exist", query_id=20250724_013940_00017_typcy)`

In the new model i wrote for metropolis, it has been failing to compile citing this error above. The above schema is not supposed to even exist and i don't know why it's trying to call it. 
You know it was working earlier before you commented that:  "there is no indication that this metropolis logic will be used heavily by other DEXs/models in the future. the macro includes a harcoded CTE which is specific to metropolis on sonic. let's remove the need for a macro and put that SQL logic in this model file directly".  that's why i had to change the whole thing and it has breaking since then and Metropolis has only functions and no events which makes it this way. 
I noticed that the [models/trades/sonic/platforms/tapio_sonic_base_trades.sql](https://github.com/duneanalytics/spellbook/blob/953de41ee44f506ee576338c546a4f60861ce8dd/models/trades/sonic/platforms/wagmi_sonic_base_trades.sql)  has function calls too and it has its own Macro built for it here >>

okolicodes7 commented:
Hello @jeff-dude
